src/Data/Array/Accelerate/Linear/Vector.hs

```diff
@@ -174,8 +178,8 @@ negated = lift1 (L.negated :: f (Exp a) -> f (Exp a))
 
 -- | Subtraction with a scalar on the right
 --
--- >>> V2 1 2 ^- 3
--- V2 (-2) (-1)
+-- >>> lift (V2 1 2 :: V2 Int) ^- 3
+-- (2,1)
```

Comment:
Actually based on the comment I think it was the implementation which was incorrect; I fixed this here: https://github.com/tmcdonell/linear-accelerate/commit/571c424ace0a0ffc4e2430eac9fe11e14cc1c7fd